zerver/actions/create_user.py

```diff
@@ -233,9 +237,12 @@ def add_new_user_history(user_profile: UserProfile, streams: Iterable[Stream]) -
             um = UserMessage(
                 user_profile=user_profile, message_id=message_id, flags=UserMessage.flags.historical
             )
-            if new_onboarding_messages_exist and message_id not in new_onboarding_message_ids:
-                um.flags |= UserMessage.flags.read
-            if not new_onboarding_messages_exist and message_id in older_message_ids:
+            if new_onboarding_messages_exist:
+                if message_id not in new_onboarding_message_ids:
+                    um.flags |= UserMessage.flags.read
+                elif message_id_to_onboarding_user_message[message_id].flags.starred.is_set:
+                    um.flags |= UserMessage.flags.starred
```

Comment:
You set `flags = OnboardingUserMessage.flags.historical` below, but that seems to be unused with this construction. Maybe that's OK?

Reply:
I see. We could in theory just have used the same `ALL_FLAGS` set from `AbstractUserMessage`, but I guess this is probably the better design in that we may in the future have other flags specific to onboarding user messages.